packages/react-components/react-drawer/src/components/Drawer/Drawer.types.ts

```diff
@@ -1,17 +1,78 @@
+import type { DialogProps, DialogSurfaceProps } from '@fluentui/react-dialog';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type DrawerSlots = {
+  /**
+   * Root slot of the Drawer.
+   * This slot is rendered as a `div` by default.
+   * Only renders if `type` is `inline`.
+   */
   root: Slot<'div'>;
+
+  /**
+   * Dialog component that renders for overlay drawer.
+   * The default is an instance of Dialog component
+   * This slot expects a compatible Dialog which will replace the default dialog component.
+   */
+  dialog?: NonNullable<Slot<DialogProps>>;
+
+  /**
+   * DialogSurface component that renders for overlay drawer.
+   * The default is an instance of DialogSurface component
+   * This slot expects a compatible DialogSurface which will replace the default dialog component.
+   */
+  dialogSurface?: NonNullable<Slot<DialogSurfaceProps>>;
 };
 
+export type DrawerSizes = 'small' | 'medium' | 'large' | 'full';
+
 /**
  * Drawer Props
  */
-export type DrawerProps = ComponentProps<DrawerSlots> & {};
+export type DrawerProps = ComponentProps<DrawerSlots> & {
+  /**
+   * Position of the drawer.
+   * @defaultvalue 'left'
+   *
+   * - 'left' - Drawer is positioned on the left side of the screen.
+   * - 'right' - Drawer is positioned on the right side of the screen.
+   */
+  position?: 'left' | 'right';
+
+  /**
+   * Type of the drawer.
+   * @defaultvalue 'overlay'
+   *
+   * - 'overlay' - Drawer is hidden by default and can be opened by clicking on the trigger.
+   * - 'inline' - Drawer is stacked with the content
+   */
+  type?: 'inline' | 'overlay';
+
+  /**
+   * Size of the drawer.
+   * @defaultvalue 'small'
+   *
+   * - 'small' - Drawer is 320px wide.
+   * - 'medium' - Drawer is 592px wide.
+   * - 'large' - Drawer is 940px wide.
+   * - 'full' - Drawer is 100vw wide.
+   * - number - Drawer is the given number of pixels wide.
```

Comment:
is this by design (the `number` part)? what if user wanted to use relative units ?

Reply:
hmm, sounds good. I'm in general always on alert when there are 2 different "apis" hidden under one. in this case the `number` part. as you mentioned the override, couldn't we remove that `number` and if one needs particular value override should be used for the sake of clean API ?

Reply:
Yeah. That's better.

Reply:
Fixed!